### PR TITLE
Python utils text

### PR DESCRIPTION
**Pull request by arturkamieniecki**
## What does this PR do?

This PR extracts generating Python symbols to external utility functions. This allows the user to get function definitions outside of reading the generated Python symbols files.

## How was this PR tested?
By comparing old and newly generated Python symbols.

### CODE REVIEW

**Comment by michalpelka** [comment]
I've rebased using the latest development on upstream.

**Review by adamdbrw — APPROVED** [review]

File: Gems/EditorPythonBindings/Code/Include/EditorPythonBindings/PythonUtility.h [hunk 1]
```diff
@@ -127,5 +138,57 @@ namespace EditorPythonBindings
...
+        class PythonBehaviorDescription
+        {
+        public:
+            //! Get Python type for behavior typeId.
+            AZStd::string_view FetchPythonTypeAndTraits(const AZ::TypeId& typeId, AZ::u32 traits);
+            AZStd::string FetchPythonTypeName(const AZ::BehaviorParameter& param);
+            AZStd::string FetchOutcomeType(const AZ::TypeId& typeId);
+
+            //! Creates a string containing bus events and documentation.
+            AZStd::string BusDefinition(const AZStd::string busName, const AZ::BehaviorEBus* behaviorEBus);
```
Comment (jhanca-robotecai) [inline Gems/EditorPythonBindings/Code/Include/EditorPythonBindings/PythonUtility.h:154]: Why not passing `busName` with ref? The same goes for other methods in this file using `AZStd::string` parameters.

File: Gems/EditorPythonBindings/Code/Include/EditorPythonBindings/PythonUtility.h [hunk 2]
```diff
@@ -127,5 +138,57 @@ namespace EditorPythonBindings
...
+
+            //! Creates a string with class or global method definition and documentation.
+            //! @param defineTooltip if true, the tooltip will be included in the definition
```
Comment (jhanca-robotecai) [inline Gems/EditorPythonBindings/Code/Include/EditorPythonBindings/PythonUtility.h:157]: I do not see the reason to document one parameter and skip the documentation of the rest. The same goes for other methods in this file.

File: Gems/EditorPythonBindings/Code/Source/PythonLogSymbolsComponent.cpp [hunk 3]
```diff
@@ -439,49 +210,23 @@ namespace EditorPythonBindings
...
-
-            // add header 
+            // add header
             AZ::u64 filesize = 0;
             AZ::IO::FileIOBase::GetInstance()->Size(fileHandle->m_handle, filesize);
-            if (filesize == 0)
-            {
-                AzFramework::StringFunc::Append(buffer, "class property():\n");
-            }
+            bool needsHeader = (filesize == 0);
```
Comment (jhanca-robotecai) [inline Gems/EditorPythonBindings/Code/Source/PythonLogSymbolsComponent.cpp:226]:
```suggestion
            bool needsHeader = (filesize == 0);
```
```suggestion
            const bool needsHeader = (filesize == 0);
```

**Review by jhanca-robotecai — COMMENTED** [review]
Browsed the code diagonally - it is very hard to read due to numerous changes caused by _clang-format_ which, I belive, should not be part of this PR.

File: Gems/EditorPythonBindings/Code/Include/EditorPythonBindings/PythonUtility.h (hunk 1 above)
Comment (spham-amzn) [inline Gems/EditorPythonBindings/Code/Include/EditorPythonBindings/PythonUtility.h:154]: nit: Should pass `busName` as a string_view instead.
```suggestion
            AZStd::string BusDefinition(const AZStd::string_view busName, const AZ::BehaviorEBus* behaviorEBus);
```

File: Gems/EditorPythonBindings/Code/Include/EditorPythonBindings/PythonUtility.h [hunk 4]
```diff
@@ -127,5 +138,57 @@ namespace EditorPythonBindings
...
+            AZStd::string MethodDefinition(
+                const AZStd::string methodName,
```
Comment (spham-amzn) [inline Gems/EditorPythonBindings/Code/Include/EditorPythonBindings/PythonUtility.h:159]:
```suggestion
                const AZStd::string_view methodName,
```

File: Gems/EditorPythonBindings/Code/Include/EditorPythonBindings/PythonUtility.h [hunk 5]
```diff
@@ -127,5 +138,57 @@ namespace EditorPythonBindings
...
+                bool defineTooltip = false,
+                bool defineDebugDescription = false);
+
+            //! Creates a string with class definition and documentation.
+            //! @param defineProperties if true, the properties will be included in the definition
+            //! @param defineMethods if true, the methods will be included in the definition
+            //! @param defineTooltip if true, the tooltip will be included in the definition
+            AZStd::string ClassDefinition(
+                const AZ::BehaviorClass* behaviorClass,
+                const AZStd::string className,
```
Comment (spham-amzn) [inline Gems/EditorPythonBindings/Code/Include/EditorPythonBindings/PythonUtility.h:171]:
```suggestion
                const AZStd::string_view className,
```

File: Gems/EditorPythonBindings/Code/Include/EditorPythonBindings/PythonUtility.h [hunk 6]
```diff
@@ -127,5 +138,57 @@ namespace EditorPythonBindings
...
+                bool defineMethods = true,
+                bool defineTooltip = false);
+
+            //! Creates a property definition
+            AZStd::string PropertyDefinition(
+                AZStd::string_view propertyName, int level, const AZ::BehaviorProperty& property, const AZ::BehaviorClass* behaviorClass);
+
+            AZStd::string GlobalPropertyDefinition(
+                const AZStd::string moduleName,
+                const AZStd::string propertyName,
```
Comment (spham-amzn) [inline Gems/EditorPythonBindings/Code/Include/EditorPythonBindings/PythonUtility.h:182]:
```suggestion
                const AZStd::string_view moduleName,
                const AZStd::string_view propertyName,
```

**Comment by spham-amzn** [comment]
> Browsed the code diagonally - it is very hard to read due to numerous changes caused by _clang-format_ which, I belive, should not be part of this PR.

I agree with this, clang-format only changes makes it a little difficult to parse out the actual changes, and PRs should be as focused to its intent only as much as possible

**Review by jhanca-robotecai — APPROVED** [review]

**Comment by AMZN-daimini** [comment]
These changes are causing warnings that prevent the engine from compiling on my machine. I have a fix in the PR above :)

**Pull request merged**